app/controllers/assignment_invitations_controller.rb

```diff
@@ -0,0 +1,21 @@
+class AssignmentInvitationsController < ApplicationController
+  before_action :authenticate_with_pre_login_destination, only: [:show]
+
+  def show
+    @invitation = AssignmentInvitation.find_by_key!(params[:id])
+
+    if @invitation.redeem(current_user)
+      render :success, layout: false, status: 200
```

Comment:
right now I'm just returning a layout with `success` for `failed`, obviously these need to be more descriptive. For success what do you think about maybe giving the repo url. For failure, probably just better copy.